### PR TITLE
All labor admins and supervisors should be able to evaluate any student in their department.

### LINKED ISSUE #363: Supervisor should be able to evaluate any student in their labor department.
User Story: As a Supervisor within my department, I should be able to evaluate any labor student within the department, so that the work can be distributed among all departmental supervisors.

Acceptance Criteria: 
- A supervisor should be able to view and evaluate labor students within their department. 
- A labor admin should be able to evaluate any student in their labor department.

Sub-issue of #359

### PR DESCRIPTION
**Pull request by tylerpar99**
This PR fixes #363 

<h3>Summary: </h3><p>This PR will allow all Labor admins and supervisors to evaluate any student in their department. The logic works when a supervisor or Labor Admin is assigned to multiple departments. </p>

<h3>To Test:</h3> <p>You will by default be "heggens" and he by default is a labor admin. While logged in as him make sure you are using staging and prod data. Open up the term "AY 2021-2022" and make sure evaluations are open and the term is open. You should now be able to go to the Supervisor Portal and search with terms "AY 2021-2022" and Department "Computer Science" and click "Labor History" on the Supervisor Portal table and the evaluate button should be visible. Now take away admin privileges and you should be able to click on any labor history in your department that you are not a direct supervisor for and the evaluate button should also be visible.</p>

<h3>Note:</h3><p>This branch is branched off of PR #386 which is branched off of PR #383.</p>

### CODE REVIEW

**Review by sreynit02 — COMMENTED** [review]
- Links: I think the brower's back button,  "Save for later" and "Submit Final Evaluation" button should lead the user back to the result table of the search. If a supervisor has multiple students to evaluate they will have to search so many times to complete all the evaluations.

**Review by sreynit02 — CHANGES_REQUESTED** [review]
Overall functionality works great. See the comments above for changing links suggestion.

File: app/static/js/supervisorPortal.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
 });
 
+function parseCookie(str){
+    cookieArray = str.split('=')
+    return cookieArray[1];
+  };
 
-
-function runformSearchQuery() {
+function runformSearchQuery(newData='', cookie) {
```
Comment (AndersonStettner) [inline app/static/js/supervisorPortal.js]: capitalize f of form?

**Review by BrianRamsay — APPROVED** [review]

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
When you run into the error `app.logic.userInsertFunctions.InvalidUserException: B00291242 not found in Tracy database` it completely hoses the form search. We need to handle that more gracefully, so that the record is either marked or dropped, rather than the whole table throwing a javascript error at the user. My search was for Spring 2022. The fourth page throws an error.

**Comment by tylerpar99** [comment]
> When you run into the error `app.logic.userInsertFunctions.InvalidUserException: B00291242 not found in Tracy database` it completely hoses the form search. We need to handle that more gracefully, so that the record is either marked or dropped, rather than the whole table throwing a javascript error at the user. My search was for Spring 2022. The fourth page throws an error.

Talked it over and we agreed it is a development-specific issue.

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**